pkg/bff/collaborators.go

```diff
@@ -0,0 +1,63 @@
+package bff
+
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/rs/zerolog/log"
+	"github.com/trisacrypto/directory/pkg/bff/api/v1"
+	"github.com/trisacrypto/directory/pkg/bff/db/models/v1"
+)
+
+// AddCollaborator creates a new collaborator with the email address in the request.
+// The endpoint adds the collaborator to the organization record associated with the
+// user and sends a verification email to the provided email address, so the user must
+// have the update:collaborators permission.
+func (s *Server) AddCollaborator(c *gin.Context) {
+	var (
+		err          error
+		request      *models.Collaborator
+		collaborator *models.Collaborator
+		org          *models.Organization
+	)
+
+	// Fetch the organization from the claims
+	if org, err = s.OrganizationFromClaims(c); err != nil {
+		log.Error().Err(err).Msg("could not fetch organization from claims")
+		c.JSON(http.StatusInternalServerError, "could not add collaborator")
+		return
+	}
+
+	// Unmarshal the collaborator from the POST request
+	request = &models.Collaborator{}
+	if err = c.ShouldBind(request); err != nil {
+		log.Warn().Err(err).Msg("could not bind request")
+		c.JSON(http.StatusBadRequest, api.ErrorResponse(err))
+		return
+	}
+
+	// Add the collaborator to the organization
+	if err = AddOrganizationCollaborator(org, request.Email); err != nil {
```

Comment:
I suppose any validation for the collaborator model will happen here? 

Any reason we're just passing the email and not the whole Collaborator struct?

Reply:
In the first design I was passing in the email to endpoint instead of the struct, this is a remnant of that but I think it does make sense to pass in the struct here and have `AddOrganizationCollaborator` handle the validation.